machine-runner/tests/esm/runner.test.ts

```diff
@@ -801,27 +801,38 @@ describe('machine as async generator', () => {
     it("should not affect parent's destroyed status", async () => {
       const r = new Runner(On, { toggleCount: 0 })
       const machine = r.machine
-      const cloned = machine.noAutoDestroy()
+      const cloned1 = machine.noAutoDestroy()
+      const cloned2 = machine.noAutoDestroy()
+      const cloned3 = machine.noAutoDestroy()
 
       await r.feed([{ type: ToggleOff.type }], true)
       // peek should come before next
-      const cres1 = await cloned.peek()
+      const cres1 = await cloned1.peek()
+      const cres2 = await cloned2.peek()
+      const cres3 = await cloned3.peek()
       const mres1 = await machine.next()
       expect(mres1.done).toBeFalsy()
       expect(cres1.done).toBeFalsy()
+      expect(cres2.done).toBeFalsy()
+      expect(cres3.done).toBeFalsy()
 
       await r.feed([{ type: ToggleOn.type }], true)
 
       // attempt to kill
-      cloned.return?.()
-      cloned.throw?.()
+      cloned1.return?.()
+      cloned2.throw?.()
+      cloned3.destroy()
 
       // peek should come before next
-      const cres2 = await cloned.peek()
-      const mres2 = await machine.next()
-
-      expect(mres2.done).toBeFalsy()
-      expect(cres2.done).toBeTruthy()
+      const cres1_2 = await cloned1.peek()
+      const cres2_2 = await cloned1.peek()
+      const cres3_2 = await cloned1.peek()
```

Comment:
isn’t the point of `.noAutoDestroy()` (since my changes) that this should also work with `.next()`?

Reply:
ah, my bad. those should be cloned1, cloned2, and cloned 3